File: 191.number-of-1-bits.py
#
# @lc app=leetcode id=191 lang=python3
#
# [191] Number of 1 Bits
#

# Time:  O(1)
# Space: O(1)

# @lc code=start
class Solution:
    def hammingWeight(self, n: int) -> int:
        res = 0
        for i in range(32):
            res += n & 1
            n >>= 1
        return res

        res = 0
        while n:
            res += 1
            n &= n - 1
        return res


# @lc code=end

File: 191.number-of-1-bits.go
/*
 * @lc app=leetcode id=191 lang=golang
 *
 * [191] Number of 1 Bits
 */
package leetcode

// @lc code=start
func hammingWeight(num uint32) int {
	res := 0
	for num > 0 {
		res++
		num &= num - 1
	}
	return res
}

// @lc code=end
